pallets/elections/src/impls.rs

```diff
@@ -253,10 +253,8 @@ where
         <T as Config>::SessionManager::new_session(new_index);
         // new session is always called before the end_session of the previous session
         // so we need to populate reserved set here not on start_session nor end_session
```

Comment:
Added the comment with this. Not sure though if the place is correct but is it now clearer what is happening?

Reply:
Yes -- this looks great now. The numbering is a little off 1, 2, 4, 6, but otherwise it's very informative. Is it really the case though that all `end_session(S)`,  `start_session(S + 1)`, `new_session(S + 2)` happen in a single block? I don't remember this well, but I would expect at least `start_session(S+1)` to happen in the next block?

Reply:
Yep, everything is in this function here: https://github.com/paritytech/substrate/blob/bc7a1e6c19aec92bfa247d8ca68ec63e07061032/frame/session/src/lib.rs#L636

Reply:
Awesome! Thanks.